includes/admin/class-wc-rest-payments-settings-option-controller.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * Class WC_REST_Payments_Settings_Option_Controller
+ *
+ * @package WooCommerce\Payments\Admin
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * REST controller for settings options.
+ */
+class WC_REST_Payments_Settings_Option_Controller extends WC_Payments_REST_Controller {
+
+	/**
+	 * List of allowed option keys that can be updated via the REST API.
+	 *
+	 * @var array
+	 */
+	private const ALLOWED_OPTIONS = [
+		'wcpay_frt_discover_banner_settings',
+		'wcpay_multi_currency_setup_completed',
+		'woocommerce_dismissed_todo_tasks',
+		'woocommerce_remind_me_later_todo_tasks',
+		'woocommerce_deleted_todo_tasks',
+		'wcpay_fraud_protection_welcome_tour_dismissed',
+		'wcpay_capability_request_dismissed_notices',
+		'wcpay_onboarding_eligibility_modal_dismissed',
+		'wcpay_connection_success_modal_dismissed',
+		'wcpay_next_deposit_notice_dismissed',
+		'wcpay_duplicate_payment_method_notices_dismissed',
+		'wcpay_exit_survey_dismissed',
+		'wcpay_instant_deposit_notice_dismissed',
+		'wcpay_date_format_notice_dismissed',
+	];
+
+	/**
+	 * Endpoint path.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'payments/settings';
+
+	/**
+	 * Configure REST API routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<option_key>[a-zA-Z0-9_-]+)',
+			[
+				'methods'             => WP_REST_Server::EDITABLE,
+				'callback'            => [ $this, 'update_option' ],
+				'permission_callback' => [ $this, 'check_permission' ],
+				'args'                => [
+					'option_key' => [
+						'required'          => true,
+						'validate_callback' => [ $this, 'validate_option_key' ],
+					],
+					'value'      => [
+						'required' => true,
+					],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Verify access.
+	 *
+	 * @return bool
+	 */
+	public function check_permission(): bool {
+		return current_user_can( 'manage_woocommerce' );
+	}
+
+	/**
+	 * Validate the option key.
+	 *
+	 * @param string $option_key The option key to validate.
+	 * @return bool
+	 */
+	public function validate_option_key( string $option_key ): bool {
+		return in_array( $option_key, self::ALLOWED_OPTIONS, true );
+	}
+
+	/**
+	 * Update the option value.
+	 *
+	 * @param WP_REST_Request $request The request object.
+	 * @return WP_REST_Response
+	 */
+	public function update_option( WP_REST_Request $request ) {
+		$option_key = $request->get_param( 'option_key' );
+		$value      = $request->get_param( 'value' );
```

Comment:
I'm wondering whether we can force the value to either be a boolean, or an array with boolean values, based on the options that are currently being saved. 🤔

Reply:
I've reviewed more options and here how is their value format looks like: 

| Option Name | Value Format | Value Example |
|------------|--------------|---------------|
| `woocommerce_dismissed_todo_tasks`, `woocommerce_remind_me_later_todo_tasks`, `woocommerce_deleted_todo_tasks` | `array<string>` | `['reconnect-wpcom']` |
| `wcpay_capability_request_dismissed_notices` | `array<string, boolean>` | `['jcb' => true]` |
| `wcpay_duplicate_payment_method_notices_dismissed` | `array<string, array<string>>` | `['bancontact' => ['woocommerce_payments', 'another_plugin']` |

Given this nature, I think we might simply check values to be either `boolean` or `array`. WDYT?

Reply:
That works, as we need a quick solution.

A `switch` like this could make a ton of sense:

```php
switch ( $request->get_param( 'option_key' ) ) {
	case 'X':
	case 'Y':
		// validate boolean;
		break;
	
	case 'Z':
		// check for array with boolean values;
		break;
	
	default:
		throw new Exception( 'Please implement validation for ' . $request->get_param( 'option_key' ) );
}
```

However, given how hard it is to get the right modal/banner to appear, you'd need to spend about an hour to check that each option works. With 14 options, you'd need a day or two to make sure that all options are validated properly. Then the reviewer would need to do the same, and that's just not feasible at the moment.

Reply:
Addressed at 84029a02281dd755e8186b34986ebdb411cd1384